artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+type ExpirationPolicy struct {
+	Enabled        *bool `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+	PasswordMaxAge *int  `yaml:"passwordMaxAge,omitempty" xml:"passwordMaxAge,omitempty"`
+	NotifyByEmail  *bool `yaml:"notifyByEmail,omitempty" xml:"notifyByEmail,omitempty"`
+	Reset          *bool `yaml:"-" xml:"-"`
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (e ExpirationPolicy) MarshalYAML() (interface{}, error) {
+	if e.Reset != nil && *e.Reset {
+		return nil, nil
+	}
+	return e, nil
+}
+
+// ResetPolicy represents the Password Reset Protection policy settings in Artifactory Security Configuration.
+// This is undocumented in YAML Configuration File.
+//
+type ResetPolicy struct {
+	Enabled               *bool `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+	MaxAttemptsPerAddress *int  `yaml:"maxAttemptsPerAddress,omitempty" xml:"maxAttemptsPerAddress,omitempty"`
+	TimeToBlockInMinutes  *int  `yaml:"timeToBlockInMinutes,omitempty" xml:"timeToBlockInMinutes,omitempty"`
+	Reset                 *bool `yaml:"-" xml:"-"`
+}
+
+func (r ResetPolicy) MarshalYAML() (interface{}, error) {
+	if r.Reset != nil && *r.Reset {
+		return nil, nil
+	}
+	return r, nil
+}
+
+// UserLockPolicy represents the User Lock Policy settings in Artifactory Security Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-Security(Generalsecurity,PasswordPolicy,LDAP,SAML,OAuth,HTTPSSO,Crowd)
+type UserLockPolicy struct {
+	Enabled       *bool `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+	LoginAttempts *int  `yaml:"loginAttempts,omitempty" xml:"loginAttempts,omitempty"`
+	Reset         *bool `yaml:"-" xml:"-"`
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (u UserLockPolicy) MarshalYAML() (interface{}, error) {
+	if u.Reset != nil && *u.Reset {
+		return nil, nil
+	}
+	return u, nil
+}
+
+// SigningKeysSetting represents the GPG Signing settings in Artifactory Security Configuration.
+// This is undocumented in YAML Configuration File.
+//
+type SigningKeysSettings struct {
+	Passphrase       *string `yaml:"passphrase,omitempty" xml:"passphrase,omitempty"`
+	KeyStorePassword *string `yaml:"keyStorePassword,omitempty" xml:"keyStorePassword,omitempty"`
+	Reset            *bool   `yaml:"-" xml:"-"`
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (s SigningKeysSettings) MarshalYAML() (interface{}, error) {
+	if s.Reset != nil && *s.Reset {
+		return nil, nil
+	}
+	return s, nil
+}
+
+// LdapSetting represents the LDAP settings in Artifactory Security Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-Security(Generalsecurity,PasswordPolicy,LDAP,SAML,OAuth,HTTPSSO,Crowd)
+type LdapSetting struct {
+	Key                      *string            `yaml:"-" xml:"key,omitempty"`
```

Comment:
I noticed we're not allowing this field to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?

Reply:
Similar to [comment](https://github.com/target/go-arty/pull/57#discussion_r436119180).